pkg/profiler/profiler.go

```diff
@@ -548,31 +559,34 @@ func (p *CgroupProfiler) profileLoop(ctx context.Context, captureTime time.Time)
 	return nil
 }
 
-func (p *CgroupProfiler) loopReport(lastProfileTakenAt time.Time, lastError error) {
+func (p *CgroupProfiler) loopReport(lastError error) {
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 
-	p.lastProfileTakenAt = lastProfileTakenAt
+	if lastError == nil {
```

Comment:
I like this, but then I think we should rename the fields to `lastSuccessfulProfileStartedAt` and `nextProfileStartedAt`.